### PR TITLE
Improve docs on how to register a custom operator for `@whereConditions`

### PR DESCRIPTION
**Pull request by lorado**
- [ ] Added or updated tests
- [x] Added Docs for all relevant versions
- [ ] Updated CHANGELOG.md

Related to #1151 

**Changes**

Added detailed description how to create and register custom operator implementation for `@whereConditions`

**Breaking changes**

no

### CODE REVIEW

File: docs/master/eloquent/complex-where-conditions.md [hunk 1]
```diff
@@ -322,3 +322,52 @@ This query would retrieve all persons, no matter if they have a role or not:
...
+
+First create a class that implements `\Nuwave\Lighthouse\WhereConditions\Operator`, e.g. `\App\GraphQL\CustomSQLOperator`.
+In following example we will extend `\Nuwave\Lighthouse\WhereConditions\SQLOperator` that actually implements 
+`\Nuwave\Lighthouse\WhereConditions\Operator` and modify result of `enumDefinition()` to include `ILIKE` operator to the
+end of the `SQLOperator` enum. 
+
+```php
+use Nuwave\Lighthouse\WhereConditions\SQLOperator;
+
+class CustomSQLOperator extends SQLOperator
```
Comment (spawnia) [inline docs/master/eloquent/complex-where-conditions.md]: I think this is quite fragile, so i would not recommend doing this to users.

File: docs/master/eloquent/complex-where-conditions.md [hunk 2]
```diff
@@ -322,3 +322,52 @@ This query would retrieve all persons, no matter if they have a role or not:
...
+after `\Nuwave\Lighthouse\WhereConditions\WhereConditionsServiceProvider::class` in your `config/app.php`. E.g. create `\App\Providers\GraphQLServiceProvider`:
+
+```php
+use App\GraphQL\CustomSQLOperator;
+use Illuminate\Support\ServiceProvider;
+use Nuwave\Lighthouse\WhereConditions\Operator;
+
+class GraphQLServiceProvider extends ServiceProvider
+{
+    public function register()
```
Comment (spawnia) [inline docs/master/eloquent/complex-where-conditions.md]:
```suggestion
    public function register(): void
```

File: docs/master/eloquent/complex-where-conditions.md (hunk 1 above)
Comment (lorado) [reply docs/master/eloquent/complex-where-conditions.md]: Ye... ok. do you have an example for me? :D

File: docs/master/eloquent/complex-where-conditions.md [hunk 3]
```diff
@@ -322,3 +322,52 @@ This query would retrieve all persons, no matter if they have a role or not:
...
+    {
+        $this->app->bind(Operator::class, CustomSQLOperator::class);
+    }
+}
+```
+
+Don't forget to register you just created service provider in `config/app.php`
```
Comment (spawnia) [inline docs/master/eloquent/complex-where-conditions.md]:
```suggestion
Don't forget to register your new service provider in `config/app.php`:
```

File: docs/master/eloquent/complex-where-conditions.md (hunk 1 above)
Comment (spawnia) [reply docs/master/eloquent/complex-where-conditions.md]: We have been discussing some ways that allow extending the `SQLOperator` more cleanly, let's hold off on our recommended solution until we implemented something better.

Doing string manipulation on the definition string is really hacky.

**Comment by spawnia** [comment]
I think the part that shows how to register custom `Operator` implementations is great, but let's not open the door on extending the `SQLOperator` until we have a nicer solution in place.

**Comment by lorado** [comment]
@spawnia how about this text?

**Comment by spawnia** [comment]
Did one pass over it, you can add final tweaks and merge then.

**Comment by lorado** [comment]
It's good. I was a little confused of packing service provider into `\App\GraphQL` namespace, as I like to put all service providers in one place, like config. But I noticed, that in the enum registering example you also used `\App\GraphQL` namespace. So I'll let it as is to stay consistent.

**Pull request merged**